packages/vega-statistics/src/regression/constant.js

```diff
@@ -0,0 +1,17 @@
+import rSquared from './r-squared';
+
+export default function (data, x, y) {
+  let mean = 0, n = 0;
+  for (const d of data) {
+    const val = y(d)
+    if (x(d) == null || val == null || isNaN(val)) continue;
+    mean += (val - mean) / ++n;
+  }
+  const predict = () => mean;
+
+  return {
+    coef: [mean],
+    predict: predict,
+    rSquared: rSquared(data, x, y, mean, predict)
```

Comment:
`rSquared` is always `0` for a constant fit. We could skip that. 
I guess this would bring efficiency at the cost of loosing consistency in the implementation when compared to other orders?

## Proof
Here's the `rSquared` implementation and with `predict` being equal to the mean being equal to `uY`, `SSE` and `SST` are equal and `1 - SSE / SST` equals zero. 
```
export default function(data, x, y, uY, predict) {
  let SSE = 0, SST = 0;

  visitPoints(data, x, y, (dx, dy) => {
    const sse = dy - predict(dx),
          sst = dy - uY;

    SSE += sse * sse;
    SST += sst * sst;
  });

  return 1 - SSE / SST;
}```

Reply:
Thanks. I updated the code.